src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1457,7 +1457,7 @@ private MethodSymbol FindEntryPoint(CancellationToken cancellationToken, out Imm
                 var viableEntryPoints = ArrayBuilder<MethodSymbol>.GetInstance();
                 foreach (var candidate in entryPointCandidates)
                 {
-                    if (!candidate.HasEntryPointSignature(this))
+                    if (!candidate.HasEntryPointSignature(this, diagnostics))
```

Comment:
Yeah, I know which bug you are talking about;  I'd be happy to write a test in this PR that fails, but fixing it will entail rewriting the entrypoint-locator, which is going to happen in the next PR.